api/v1alpha1/selfnoderemediationconfig_webhook_test.go

```
@@ -0,0 +1,141 @@
+					snrc := createSelfNodeRemediationConfigCR(item.fieldName, item.fieldValue)
+					err := snrc.ValidateUpdate(snrcOld)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring(item.errorMessage))
+				})
+			})
+		}
+
+		Context("for valid CR", func() {
+			It("should not be rejected", func() {
+				snrcOld := createDefaultSelfNodeRemediationConfigCR()
+				snrc := createDefaultSelfNodeRemediationConfigCR()
+				err := snrc.ValidateUpdate(snrcOld)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+		})
+
+	})
+
+	Describe("parsing time duration string to milliseconds", func() {
+		Context("for the string 10ms", func() {
+			It("should return 10", func() {
+				Expect(toMS("10ms")).To(Equal(int64(10)))
+			})
+		})
+
+		Context("for the string 10s", func() {
+			It("should return 10000", func() {
+				Expect(toMS("10s")).To(Equal(int64(10000)))
+			})
+		})
+
+	})
+})
+
+func createDefaultSelfNodeRemediationConfigCR() *SelfNodeRemediationConfig {
+	snrc := &SelfNodeRemediationConfig{}
+	snrc.Name = "test"
+	snrc.Namespace = "default"
+
+	//default values for time fields
+	snrc.Spec.PeerApiServerTimeout = &metav1.Duration{Duration: 5 * time.Second}
+	snrc.Spec.ApiServerTimeout = &metav1.Duration{Duration: 5 * time.Second}
+	snrc.Spec.PeerDialTimeout = &metav1.Duration{Duration: 5 * time.Second}
+	snrc.Spec.PeerRequestTimeout = &metav1.Duration{Duration: 5 * time.Second}
+	snrc.Spec.ApiCheckInterval = &metav1.Duration{Duration: 15 * time.Second}
+	snrc.Spec.PeerUpdateInterval = &metav1.Duration{Duration: 15 * time.Minute}
+
+	return snrc
+}
+
+func createSelfNodeRemediationConfigCR(fieldName string, value string) *SelfNodeRemediationConfig {
+	snrc := createDefaultSelfNodeRemediationConfigCR()
+
+	//set the field tested
+	setFieldValue(snrc, fieldName, value)
+
+	return snrc
+}
+
+func setFieldValue(snrc *SelfNodeRemediationConfig, fieldName string, value string) {
+	d, err := time.ParseDuration(value)
+	if err != nil {
```

replace with d, _ if you choose to ignore the error (I think it can be acceptable in some case of test  code).